apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts

```diff
@@ -82,3 +83,19 @@ export function getPersistedOptions(): NamedImportMap {
     return namedImportMap;
   }
 }
+
+export function addExistingCustomFields(importMap: NamedImportMap, customFields: CustomFields): NamedImportMap {
+  const custom = importMap.custom;
```

Comment:
question: is this just a convenience rellabelling? 
did you check that we are mutating the original `importMap`?